src/Compilers/Core/Portable/CodeGen/ITokenDeferral.cs

```diff
@@ -21,6 +21,8 @@ internal interface ITokenDeferral
 
         Cci.IFieldReference GetArrayCachingFieldForConstants(ImmutableArray<ConstantValue> constants, Cci.IArrayTypeReference arrayType, SyntaxNode syntaxNode, DiagnosticBag diagnostics);
 
+        Cci.IFieldReference? GetFieldForDataString(string text, ImmutableArray<byte> data, SyntaxNode syntaxNode, DiagnosticBag diagnostics);
```

Comment:
>GetFieldForDataString [](http://example.com/codeflow?start=29&length=21)

It feels like the "GetFieldForDataString" name is somewhat confusing. What is a "data string" for which we are requesting a field? It looks like we are requesting a static field that is supposed to store the value of the ```text``` parameter during at runtime. Therefore, "GetFieldForStringValue" feels better.

Reply:
Also, If we expect this function to fail to produce a field under some conditions, consider adding "Try" prefix to the name of the function.